ingest_missing.py

```diff
@@ -0,0 +1,169 @@
+    startDate = datetime.strptime(startDate, '%Y-%m-%d %H:%M:%S')
+    endDate = datetime.strptime(endDate, '%Y-%m-%d %H:%M:%S')
+    
+    date_times = [startDate.strftime('%Y%m%dT%H')]
+    date_time = startDate
+    
+    while date_time < endDate:
+        date_time += timedelta(hours=1)
+        date_times.append(date_time.strftime('%Y%m%dT%H'))
+
+    return date_times
+
+
+def playback(refdes, event_url, particle_url, missing_dates):
+    """
+    Put together the directory glob and other options for the playback command.
+    This will then playback cabled data.
+    """
+    main_directory = '/rsn_cabled/rsn_data/DVT_Data'
+    driver = get_driver(refdes)
+    reader_list = get_reader_type(refdes)
+    node = refdes.split('-')[1].lower()
+    instrument = refdes.split('-')[3]
+    signal.signal(signal.SIGALRM, timeout_handler)
+
+    for reader in reader_list:
+        for date in missing_dates:
+            directory = '/'.join([main_directory, node, instrument])
+            directory += '*'.join(['', date, ''])
+
+            # Check to see if this particular file exists before executing callback
+            if glob.glob(directory):
+                playback_command = ' '.join(['playback', reader, driver, refdes, event_url, particle_url, directory])
+                logging.info("%s", playback_command)
+
+                # Some playback reader types may not ingest the data at all.
+                # Time out after 90 seconds an contininue to the next reader type or
+                # next data.
+                signal.alarm(90)    
+                try:
+                    call(playback_command, shell=True)
+                except TimeoutException:
+                    logging.warning('%s Took more than 120 seconds. Timing out this ingestion.', playback_command)
+                    continue 
+                else:
+                    signal.alarm(0)
+
+
+def main(args):
+    logging.getLogger().setLevel(logging.INFO)
+    if args['<event_url>'] and args['<particle_url>'] and args['<server>']:
+        Cabled.cabled_drivers_raw = request_cabled_raw()
+        event_url = args['<event_url>']
+        particle_url = args['<particle_url>']
+        server = args['<server>']
+        refdes_list = list_missing_dates.get_refdes_list(server)
+        missing_data_dict = {}
+        
+        for refdes in refdes_list:
+            if is_cabled(refdes):
+                missing_data_list = list_missing_dates.get_missing_data_list(refdes, server) # returns tuple
+                if missing_data_list:
+                    missing_data_dict[refdes] = missing_data_list
+        
+        for refdes, missing_dates in missing_data_dict.iteritems():
+            for date in missing_dates:
+                dates = date_list(date[0], date[1])
+                playback(refdes, event_url, particle_url, dates)
+
+if __name__ == "__main__":
+    args = docopt(__doc__)
```

Comment:
I've not see the docopt in the conditional statement before, it's usually been in the main function.